applications/MeshMovingApplication/trilinos_extension/CMakeLists.txt

```diff
@@ -15,6 +15,7 @@ file(
 ## Python module
 pybind11_add_module( KratosMeshMovingTrilinosExtension MODULE THIN_LTO ${KRATOS_MESH_MOVING_TRILINOS_EXTENSION_PYTHON_INTERFACE_SOURCES} )
 
+target_include_directories(KratosMeshMovingTrilinosExtension SYSTEM PUBLIC ${TRILINOS_INCLUDE_DIR})
```

Comment:
Are you sure this is necessary? The `KratosTrilinosCore` target already does this, IMO it should propagate. Since it worked so far I think the problem is somewhere else

Reply:
Yep it is necessary,

As fas as I understand, when CMake resolves the include and linking directories, it takes into account the list of include and linking dirs added with `target_include_directories` from all the projects it has configured during an execution.

Before this changes, we were compiling trilinos along with all other major apps, hence, when including KratosTrillinosCore as a target dependency of, for instance, KratosFluidDynamicsTrilinosExtension, the include was propagated.

After the changes, the compilation of applications can be isolated in different runs, provided that the library exists. This would allow you do to something like:

```
-DKRATOS_ROOT="/already/compiled/kratos"
-DKRATOS_FLUID_DYNAMICS_ROOT="/already/compiled/fluidDynamics"
etc...
```

Which is similar to what is happening with the tiering (the only difference is that the linking directory is currently hardcoded to the install dir, but that could change in the future). With this situation the problem comes the moment the library is added as a linking dependency without having information about how it was created. Then the `target_include_directories` was never executed and cmake has no idea on where to find the headers from trilinos (which are used mainly in the python interface)

Reply:
To be honest, now that I wrote the response I realize that mostly the TrilinosExtensions are dependent on Trilinos includes alone, not TrilinosApplication. Probably linking the whole application into the extension is not the optimal solution....